contentcuration/contentcuration/frontend/shared/data/serverSync.js

```diff
@@ -306,6 +306,16 @@ if (process.env.NODE_ENV !== 'production' && typeof window !== 'undefined') {
 async function handleChanges(changes) {
   changes.map(applyResourceListener);
   const syncableChanges = changes.filter(isSyncableChange);
+  // Here we are handling changes triggered by Dexie Observable
+  // this is listening to all of our IndexedDB tables, both for resources
+  // and for our special Changes table.
+  // Here we look for any newly created changes in the changes table, which may require
+  // a sync to send them to the backend - this is particularly important for
+  // MOVE, COPY, PUBLISH, and SYNC changes where we may be executing them inside an IGNORED_SOURCE
+  // because they also make UPDATE and CREATE changes that we wish to make in the client only.
+  const newChangeTableEntries = changes.some(
+    c => c.table === CHANGES_TABLE && c.type === CHANGE_TYPES.CREATED
```

Comment:
So for @ozer550's websocket implementation, he should also check that `synced` on the underlying object is also `false` before sending a change to the backend? Unless I'm misunderstanding, changes coming from the backend would also cause this Dexie create change?

Reply:
Actually, we don't save the incoming changes to IndexedDB, do we? But we would apply them of course, so I'm just wondering how we ignore those changes that occur from incoming changes

Reply:
I forgot about `changes.filter(isSyncableChange);` which would prevent the generation of changes from incoming events being applied.